File: unittest/scripts/auto/py_shell/validation/plugin_cli_integration_norecord.py
#@<OUT> CLI --help
The following objects provide command line operations:

   cli_tester
      CLI Integration Testing Plugin

   cluster
      Represents an InnoDB cluster.

   dba
      InnoDB cluster and replicaset management functions.

   rs
      Represents an InnoDB ReplicaSet.

   shell
      Gives access to general purpose functions and properties.

   util
      Global object that groups miscellaneous tools like upgrade checker and
      JSON import.

#@<OUT> CLI plugin --help
The following object provides command line operations at 'cli_tester':

   emptyChild
      Empty object, exposes no functions but child does.

The following operations are available at 'cli_tester':

   test
      Testing cmd line args.

#@<OUT> CLI plugin function --help
NAME
      test - Testing cmd line args.

SYNTAX
      cli_tester test <stritem> <strlist> --namedstr=<str> <options>

WHERE
      stritem: String - string parameter.
      strlist: Array - string list parameter.

OPTIONS
--namedstr=<str>
            String named parameter (for being after a list parameter).

--str=<str>
            String option.

--strlist=<str list>
            String list option.

#@<OUT> Interactive plugin function --help
ERROR: Invalid operation for cli_tester object: test-interactive

#@<OUT> Interactive plugin function help
NAME
      testInteractive - Testing interactive function.

SYNTAX
      cli_tester.testInteractive()

#@<OUT> CLI plugin nested child --help
The following object provides command line operations at 'cli_tester emptyChild':

   grandChild
      Grand child object exposing a function.

#@<OUT> CLI plugin nested grand child --help
The following operations are available at 'cli_tester emptyChild grandChild':

   grand-child-function
      Testing cmd line args in nested objects.

#@<OUT> CLI plugin nested grand child function --help
The following operations are available at 'cli_tester emptyChild grandChild':

   grand-child-function
      Testing cmd line args in nested objects.

#@<OUT> Test string list handling
String Parameter: 1,2,3,4,5
String List Parameter Item: 1
String List Parameter Item: 2
String List Parameter Item: 3
String List Parameter Item: 4
String List Parameter Item: 5
String Named Item: 1,2,3,4,5
String Option: 1,2,3,4,5
String List Option Item: 1
String List Option Item: 2
String List Option Item: 3
String List Option Item: 4
String List Option Item: 5

#@<OUT> Test string list quoted args for interpreter
String Parameter: 1,2,3,4,5
String List Parameter Item: 1,2
String List Parameter Item: 3
String List Parameter Item: 4,5
String Named Item: 1,2,3,4,5
String Option: 1,2,3,4,5
String List Option Item: 1,2
String List Option Item: 3
String List Option Item: 4,5

#@<OUT> Escaped comma in list parsing
String Parameter: 1,2,3,4,5
String List Parameter Item: 1,2
String List Parameter Item: 3
String List Parameter Item: 4,5
String Named Item: 1,2,3,4,5
String Option: 1,2,3,4,5
String List Option Item: 1,2
String List Option Item: 3
String List Option Item: 4,5

#@<OUT> Escaped quoting: \", \'
String Parameter: "1",2,3,4,5
String List Parameter Item: 1
String List Parameter Item: 2
String List Parameter Item: 3
String List Parameter Item: 4
String List Parameter Item: 5
String Named Item: 1,2,"3",4,5
String Option: 1,2,"3",4,5
String List Option Item: 1
String List Option Item: 2
String List Option Item: 3
String List Option Item: 4
String List Option Item: 5
String Parameter: '1',2,3,4,5
String List Parameter Item: 1
String List Parameter Item: 2
String List Parameter Item: 3
String List Parameter Item: 4
String List Parameter Item: 5
String Named Item: 1,2,'3',4,5
String Option: 1,2,'3',4,5
String List Option Item: 1
String List Option Item: 2
String List Option Item: 3
String List Option Item: 4
String List Option Item: 5

#@<OUT> Escaped equal: \=
String Parameter: =1,2,3,4,5
String List Parameter Item: 1=2
String List Parameter Item: 3
String List Parameter Item: 4=5
String Named Item: =1,2,3,4=5
String Option: =1,2,3,4=5
String List Option Item: 1=2
String List Option Item: 3
String List Option Item: 4=5

#@<OUT> CLI calling plugin nested grand child function
Unique Parameter: Success!!

File: unittest/scripts/auto/js_shell/validation/plugin_cli_integration_norecord.js
//@<OUT> CLI -- --help
The following objects provide command line operations:

   cluster
      Represents an InnoDB Cluster.

   clusterset
      Represents an InnoDB ClusterSet.

   customObj
      Dummy global object

   dba
      InnoDB Cluster, ReplicaSet, and ClusterSet management functions.

   rs
      Represents an InnoDB ReplicaSet.

   shell
      Gives access to general purpose functions and properties.

   util
      Global object that groups miscellaneous tools like upgrade checker and
      JSON import.

//@<OUT> CLI customObj showInfo --help
NAME
      show-info - Retrieves personal information

SYNTAX
      customObj show-info <name> <age> <adult> <hobbies> --address=<str>
      <family>

WHERE
      name: String - Name of person.
      age: Integer - Age of person.
      adult: Bool - If person is an adult.
      hobbies: Array - List of hobbies.

OPTIONS
--address=<str>
            Adress location.

--wifeName=<str>
            Wife name.

--sonName=<str>
            Son name.

--dogName[:<type>]=<value>
            Dog name.

//@<OUT> WL14297 - TSFR_2_6_1
My name is John
Next year I will be 31
I am an adult
My hobbies are:
run
tennis
My address is Str.123
My family is:
Tess
Jack
Nick

//@<OUT> WL14297 - TSFR_2_6_2
ERROR: The following required option is missing: --address

//@<OUT> WL14297 - TSFR_5_1_3_3 - Testing list params can be passed as anonymous args, comma separated list of values or json array
a
b
c
a
b
c
a
b
c
a
b
c
a
b
c

//@<OUT> WL14297 - TSFR_6_1_2 - Testing list option can be passed as comma separated list of values, json array or repeated list definition
test
{
    "testOption": [
        "a", 
        "b", 
        "c"
    ]
}
test
{
    "testOption": [
        "a", 
        "b", 
        "c"
    ]
}
test
{
    "testOption": [
        "a", 
        "b", 
        "c"
    ]
}
test
{
    "testOption": [
        "a", 
        "b", 
        "c"
    ]
}
test
{
    "testOption": [
        "a", 
        "b", 
        "c"
    ]
}

//@<OUT> WL14297 - TSFR_8_2_1 - Testing dictionary with no options takes options that don't belong to other dictionary
{
    "testOther": 23, 
    "yetAnother": "sample"
}
{
    "testOption": 45
}

//@<OUT> WL14297 - TSFR_4_1_1 - Testing JSON Escaping is honored
"test"
{
    "optionA": "\"testOption\""
}

